src/basic/calendarspec.c

```diff
@@ -265,21 +266,15 @@ static void format_chain(FILE *f, int space, const CalendarComponent *c, bool us
         }
 
         assert(c->value >= 0);
-        if (!usec)
-                fprintf(f, "%0*i", space, c->value);
-        else if (c->value % USEC_PER_SEC == 0)
-                fprintf(f, "%0*i", space, (int) (c->value / USEC_PER_SEC));
-        else
-                fprintf(f, "%0*i.%06i", space, (int) (c->value / USEC_PER_SEC), (int) (c->value % USEC_PER_SEC));
-
-        if (c->repeat > 0) {
-                if (!usec)
-                        fprintf(f, "/%i", c->repeat);
-                else if (c->repeat % USEC_PER_SEC == 0)
-                        fprintf(f, "/%i", (int) (c->repeat / USEC_PER_SEC));
-                else
-                        fprintf(f, "/%i.%06i", (int) (c->repeat / USEC_PER_SEC), (int) (c->repeat % USEC_PER_SEC));
-        }
+
+        fprintf(f, "%0*i", space, c->value / d);
+        if (c->value % d)
```

Comment:
so far, we avoided relying on C's downgrade-to-bool feature for if checks like these unless the variable tested is actually a bool (or at least bool-like) or a pointer. For numeric we instead preferred numeric comparisions, i.e. == 0 or != 0 or > 0 or something like that. Any chance you can change that here?